### PR TITLE
Integrate with Payment connection API

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Integrate new direct debit payment connection API.

## Justify why they are needed

Remove dependency on old payment connection API and web onboarding.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2396** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2396" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2396?utm_source=stack-comment).

File: apps/store/src/pages/payment-success.tsx
Comment (gustaveen) [inline apps/store/src/pages/payment-success.tsx:1]: Trying to understand the flow, do we need a success page or can't the success url be the confirmation page?

File: apps/store/src/services/trustly/createTrustlyUrl.ts [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
-// TODO: refactor
-// eslint-disable-next-line @typescript-eslint/no-unused-vars,@typescript-eslint/require-await
-export const createTrustlyUrl = async ({ apolloClient }: Params) => {
-  return PLACEHOLDER_URL
+export const createTrustlyUrl = async ({ apolloClient, locale }: Params): Promise<string> => {
+  const response = await apolloClient.mutate<TrustlyInitMutation, TrustlyInitMutationVariables>({
+    mutation: TrustlyInitDocument,
+    variables: {
+      successUrl: PageLink.paymentSuccess({ locale }),
+      failureUrl: `/${locale}/payment/failure`,
```
Comment (gustaveen) [inline apps/store/src/services/trustly/createTrustlyUrl.ts:20]: Could be the same right?
```suggestion
      failureUrl: PageLink.paymentFailure({ locale }),
```

**Review by gustaveen — APPROVED** [review]
Looks good 😃 just some questions

File: apps/store/src/services/trustly/createTrustlyUrl.ts (hunk 1 above)
Comment (guilhermespopolin) [inline apps/store/src/services/trustly/createTrustlyUrl.ts:20]:
```suggestion
      failureUrl: PageLink.paymentFailure({ locale }),
```

**Review by guilhermespopolin — APPROVED** [review]
Nice, nice. Just added a small request change.

**Review by guilhermespopolin — APPROVED** [review]
Nice, nice. Just added a small request change.

**Comment by robinandeer** [comment]
@robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2396).

**Pull request merged**